sources/ClangSharp.PInvokeGenerator/Abstractions/FunctionOrDelegateDesc.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace ClangSharp.Abstractions
+{
+    public struct FunctionOrDelegateDesc<TCustomAttrGeneratorData>
+    {
+        public string AccessSpecifier;
```

Comment:
I wonder if we could track some of these via enums, both to compress space (the swath of `bool` below) and to make things like `AccessSpecifier` and `CallingConvention` less tied to what C# expects the string to be and instead something more generic that can be reused across languages.

Reply:
Would also be good to follow "best practices" for these and make the fields private and the exposed members properties with get/set (most can be auto, just may not the `bool` flags since those should be backed by a single `int32` or `int64`)